### PR TITLE
tedicross: init at 0.8.7

### PR DESCRIPTION
**Pull request by pacien**
###### Motivation for this change

This PR adds a package and a module for the [TediCross](https://github.com/TediCross/TediCross) Discord-Telegram bridge service.

###### Current upstream blockers

- [x] ~~Command-line parameters to specify path to settings and data directory: https://github.com/TediCross/TediCross/issues/101~~
- [x] ~~Do not crash when the config file isn't writable: https://github.com/TediCross/TediCross/issues/102~~
- [x] ~~Proper version release~~

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by pacien** [comment]
@GrahamcOfBorg build tedicross

**Comment by pacien** [comment]
Up.

File: nixos/modules/services/networking/tedicross.nix
Comment (aanderse) [inline nixos/modules/services/networking/tedicross.nix]: Without ever having used this program I can't say for sure, but I get the feeling [DynamicUser](https://www.freedesktop.org/software/systemd/man/systemd.exec.html) would be a better choice here. Unless you have specific reason not to I would entirely remove the `user` and `group` option in favour of DynamicUser.

File: nixos/modules/services/networking/tedicross.nix
Comment (aanderse) [inline nixos/modules/services/networking/tedicross.nix]: The default user doesn't have permission to execute either `mkdir` or `chmod` on `cfg.dataDir`.

File: nixos/modules/services/networking/tedicross.nix
Comment (aanderse) [inline nixos/modules/services/networking/tedicross.nix]: `enable = mkEnableOption "Whether to enable the TediCross Telegram-Discord bridge service.";` is the preferred way to do this.

**Review by aanderse — CHANGES_REQUESTED** [review]

**Comment by pacien** [comment]
@aanderse Thank you for your review. The PR has been amended accordingly.

File: nixos/modules/services/networking/tedicross.nix
Comment (aanderse) [reply nixos/modules/services/networking/tedicross.nix]: @pacien I made a mistake, I'm sorry!
https://github.com/NixOS/nixpkgs/blob/2e9e63195bf7e371112e95b743c1263759db6e8b/lib/options.nix#L68
So what I should have suggested is: `enable = mkEnableOption "the TediCross Telegram-Discord bridge service.";` as the "Whether to enable " bit is already there.

Comment (pacien) [reply nixos/modules/services/networking/tedicross.nix]: No problem. This is fixed.

**Comment by pacien** [comment]
Should I also add myself as a maintainer in the module file too?
I don't see a lot of modules with the `meta.maintainers` property.

**Review by aanderse — APPROVED** [review]

**Comment by aanderse** [comment]
The module LGTM :+1: 

I'm not familiar with node packaging so i'll defer to someone else on that.

**Comment by Infinisil** [comment]
You should add your package to `pkgs/development/node-packages/node-packages-v10.json` instead and then call the `generate.sh` script.

**Comment by pacien** [comment]
@Infinisil Thanks for the suggestion. The PR has been amended accordingly.

**Comment by pacien** [comment]
Rebased on master's tip.

@GrahamcOfBorg build nodePackages.tedicross

**Comment by pacien** [comment]
Amended PR to init at 0.8.5 instead of 0.8.3.

@GrahamcOfBorg build nodePackages.tedicross

File: nixos/modules/services/networking/tedicross.nix
Comment (Infinisil) [inline nixos/modules/services/networking/tedicross.nix]: It might be better to define an `environment = mkOption { type = types.attrsOf types.str; ... }`, then you don't need the file indirection

File: nixos/modules/services/networking/tedicross.nix
Comment (Infinisil) [inline nixos/modules/services/networking/tedicross.nix]: `baseNameOf` is in the global namespace, so you can remove the `builtins.`

File: nixos/modules/services/networking/tedicross.nix
Comment (aanderse) [reply nixos/modules/services/networking/tedicross.nix]: Unless this stores sensitive data like passwords, private tokens, etc..., in which case what you're currently doing is better.

Comment (Infinisil) [reply nixos/modules/services/networking/tedicross.nix]: Ohh right, never mind this then

File: nixos/modules/services/networking/tedicross.nix
Comment (pacien) [reply nixos/modules/services/networking/tedicross.nix]: Amended.

**Comment by pacien** [comment]
Amended PR to init at 0.8.7 instead of 0.8.5.

@GrahamcOfBorg build nodePackages.tedicross

**Comment by Infinisil** [comment]
Oh one last thing I missed: Can you split it into 2 commits, one for the package, one for the NixOS module? After that I'll merge it

**Comment by pacien** [comment]
@Infinisil wrote:
> Oh one last thing I missed: Can you split it into 2 commits, one for the package, one for the NixOS module? After that I'll merge it

Sure. It's done.

**Pull request merged**
